src/catalog/catalog_cache.cpp

```
@@ -157,5 +159,76 @@ std::shared_ptr<IndexCatalogObject> CatalogCache::GetCachedIndexObject(
   return nullptr;
 }
 
+/*@brief   insert sequence catalog object into cache
+ * @param   sequence_object
+ * @return  false only if sequence already exists in cache
+ */
+bool CatalogCache::InsertSequenceObject(
+    std::shared_ptr<SequenceCatalogObject> sequence_object) {
+  if (!sequence_object || sequence_object->seq_oid == INVALID_OID) {
+    return false;  // invalid object
+  }
+
+  std::size_t hash_key = GetHashKey(sequence_object->seq_name,
+                                    sequence_object->db_oid);
+
+  // check if already in cache
+  if (sequence_objects_cache.find(hash_key) !=
+          sequence_objects_cache.end()) {
```

Do you want to throw an exception for this case?